unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs

```diff
@@ -118,38 +124,42 @@ private async UniTask LoadAndDisplayEquippedWearablesAsync(UserProfile userProfi
         private void LoadAndShowOwnedWearables(UserProfile userProfile)
         {
             view.SetCollectibleWearablesLoadingActive(true);
-            wearableCatalogBridge.RequestOwnedWearables(userProfile.userId)
-                                 .Then(wearables =>
-                                  {
-                                      string[] wearableIds = wearables.GroupBy(i => i.id).Select(g => g.First().id).Take(MAX_NFT_COUNT).ToArray();
-                                      userProfile.SetInventory(wearableIds);
-                                      loadedWearables.AddRange(wearableIds);
-                                      var containedWearables = wearables.GroupBy(i => i.id).Select(g => g.First()).Take(MAX_NFT_COUNT)
-                                         .Where(wearable => wearableCatalogBridge.IsValidWearable(wearable.id));
-                                      view.SetCollectibleWearables(containedWearables.ToArray());
-                                      view.SetCollectibleWearablesLoadingActive(false);
-                                  })
-                                 .Catch(Debug.LogError);
+            wearablesPromise = wearableCatalogBridge.RequestOwnedWearables(userProfile.userId).Then(wearables =>
+            {
+                string[] wearableIds = wearables.GroupBy(i => i.id).Select(g => g.First().id).Take(MAX_NFT_COUNT).ToArray();
+                userProfile.SetInventory(wearableIds);
+                loadedWearables.AddRange(wearableIds);
+
+                var containedWearables = wearables.GroupBy(i => i.id)
```

Comment:
Is the GroupBy id needed? I thought we removed the "gotta send you two wearable of the same kind to represent you have two".

If needed, please, cache the group so you can reuse it in the below's LinQ query.